healthcheck/healthcheck.go

```diff
@@ -0,0 +1,216 @@
+// This package provides a simple health check server.
+//
+// The package lets you register arbitrary health check endpoints by
+// providing a name (== URL path) and a callback function.
+//
+// The health check server listens for HTTP requests on a given port;
+// probes are executed by issuing requests to the endpoints' named
+// URL paths.
+//
+// GETing the "/" path provides a list of registered endpoints, one
+// per line. GETing "/_ALL_" probes each registered endpoint sequentially,
+// returning each endpoint's path, HTTP status code and body per line.
+//
+// The package works as a "singleton" with just one server in order to
+// avoid cluttering the main program by passing handles around.
+package healthcheck
+
+import (
+	"net/http"
+	"fmt"
+	"bytes"
+)
+
+//
+const (
+	StatusOK = http.StatusOK
+	StatusServiceUnavailable = http.StatusServiceUnavailable
+)
+
+// Code wishing to get probed by the health-checker needs to provide this callback
+type CallbackFunc func () (code int, body string)
+
+// The HTTP server
+var server *http.Server
+
+// The HTTP request multiplexer
+var serveMux *http.ServeMux
+
+// List of endpoints known by the server
+var endpoints map[string]CallbackFunc
+
+// Init
+func init() {
+	// Create the request multiplexer
+	serveMux = http.NewServeMux()
+	// Initialize the enpoint list
+	endpoints = make(map[string]CallbackFunc)
+}
+
+// Configures the health check server
+//
+//  listenAddr: an address understood by http.ListenAndServe(), e.g. ":8008"
+func Configure(listenAddr string) {
+	// Create the HTTP server
+	server = &http.Server{
+		Addr:    listenAddr,
+		Handler: serveMux,
+	}
+
+	// Add default wildcard handler
+	serveMux.HandleFunc("/",
```

Comment:
Have a look at this: https://github.com/gorilla/mux
This might drastically simplify this.